crates/shared/src/http_solver.rs

```diff
@@ -160,13 +160,20 @@ impl HttpSolverApi for DefaultHttpSolverApi {
             url.query_pairs_mut()
                 .append_pair("auction_id", auction_id.to_string().as_str());
         }
+        let request_id = crate::request_id::get_task_local_storage();
+        if let Some(id) = &request_id {
+            url.query_pairs_mut().append_pair("request_id", id);
+        }
```

Comment:
This now also ties in Quasimodo logs since that is checking for the `request_id` URL query parameter [here](https://github.com/gnosis/gp-v2-solver-lib/blob/main/src/driver/Server.cpp#L159-L162).